### PR TITLE
Adding psrchive

### PR DESCRIPTION
**Pull request by demorest**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by demorest** [comment]
OK, this appears to be building correctly now.  If someone could review it that would be great, thanks!

**Comment by demorest** [comment]
@conda-forge/staged-recipes I think this is ready to go, could someone give it a review?  Thank you!

File: recipes/psrchive/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+{% set name = "psrchive" %}
+{% set version = "2021-01-18" %}
+
+package:
+    name: "{{ name | lower }}"
```
Comment (ocefpaf) [inline recipes/psrchive/meta.yaml:5]: @demorest this is minor but we've been using 2 spaces indentations for yaml for a while. That avoids long lines in multiple nested fields.

Comment (demorest) [reply recipes/psrchive/meta.yaml:5]: OK that is easy enough to change.

**Comment by demorest** [comment]
OK, meta.yaml is now 2-space indented.  And for completeness I added a couple more build requirements (make, perl).  Anything else that should be done for this one?

File: recipes/psrchive/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    - swig
+  host:
+    - cfitsio
+    - pgplot
+    - gsl
+    - python {{ python }}
+    - numpy
+    - fftw
+    - tempo2
+    - libcblas * *netlib
```
Comment (ocefpaf) [inline recipes/psrchive/meta.yaml:38]: I'm not 100% sure on the syntax here. I thought it should be *=*. @isuruf, sorry to bother you with this but can you take a quick look?

Comment (isuruf) [reply recipes/psrchive/meta.yaml:38]: Just `libcblas`

Comment (demorest) [reply recipes/psrchive/meta.yaml:38]: So this was a tricky/confusing issue for me.  The package does not directly depend on BLAS, only indirectly via GSL (it uses some of the `gsl_blas_*` functions).  Here are the various things I tried and what the results were:

1. Only `gsl` in the requirements (ie no BLAS library explicitly mentioned).  This resulted in openblas being installed during the build, and an "overlinking" error for libopenblas.
2. `libcblas` added to requirements.  Same results as above (build used openblas, overlinking error).
3. `libcblas * *netlib` in the requirements (host section only).  This resulted in the netlib BLAS being used successfully in the build, with no errors.  The test stage used openblas with no problem.

I arrived at this final version after reading https://conda-forge.org/docs/maintainer/knowledge_base.html#blas which implied builds should use netlib, although this didn't happen until I explicitly specified it.

All the results from these various attempts can be found in the failed builds above in this PR but please let me know if there is other specific information I can provide regarding this.  Suggestions on how best to solve this are certaintly welcome!

Comment (demorest) [reply recipes/psrchive/meta.yaml:38]: Hi folks, any additional thoughts on this?  TL;DR version of my last post was that I tried "just `libcblas`" and it didn't work.  Thanks!

Comment (ocefpaf) [reply recipes/psrchive/meta.yaml:38]: > All the results from these various attempts can be found in the failed builds above in this PR but please let me know if there is other specific information I can provide regarding this. Suggestions on how best to solve this are certaintly welcome!

Sorry I did not checked the commit history. Did you try Isuru's suggestion and that did not work? If not, do you mind testing it so we can check?

Comment (demorest) [reply recipes/psrchive/meta.yaml:38]: Yes, I did try that and it did not work (it was "#2" in my list above).  But it looks like I only did that test on my local system.  I'll push that change here so you guys can take a look.

Comment (ocefpaf) [reply recipes/psrchive/meta.yaml]: > I'll push that change here so you guys can take a look.

Thanks. If there is nothing we can do to fix it we can revert and merge as-is.

Comment (demorest) [reply recipes/psrchive/meta.yaml]: Glad this worked, thanks again for the help!  I'll have to look into why it failed for me locally, maybe different version of conda build or something like that.

**Pull request merged**